src/utils/string/StringUtils.js

```diff
@@ -346,16 +346,16 @@ const getChannelListInfo = (channel, selfSignUserId, selfAnonUserId) => {
   }
 
   if (channel?.type === 'topics') {
-    channelImage = channel?.channel_image;
+    channelImage = channel?.image || channel?.channel_image;
     channelName = channel?.name;
     if (!channelName || channelName === '') {
       channelName = `#${removePrefixTopic(channel?.id)}`;
     }
   }
 
   if (channel?.type === 'group') {
-    channelImage = channel?.is_custom_image ? channel?.channel_image : null;
-    if (channel?.is_custom_name) channelName = channel?.name;
+    channelImage = channel?.is_image_custom ? channel?.image || channel?.channel_image : null;
+    if (channel?.is_name_custom) channelName = channel?.name;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for assigning `channelImage` and `channelName` has been updated. However, it seems like there's a typo in the property names used for checking if the image or name is custom. In the new code, you're using `is_image_custom` and `is_name_custom`, while in the old code, it was `is_custom_image` and `is_custom_name`. Please ensure that the property names are correct.

```diff
-    channelImage = channel?.is_image_custom ? channel?.image || channel?.channel_image : null;
-    if (channel?.is_name_custom) channelName = channel?.name;
+    channelImage = channel?.is_custom_image ? channel?.image || channel?.channel_image : null;
+    if (channel?.is_custom_name) channelName = channel?.name;
```

Also, consider adding an else clause to handle cases where `channel?.is_name_custom` is false. Currently, if `channel?.is_name_custom` is false, `channelName` will not be assigned any value which might lead to unexpected behavior.

```diff
-    if (channel?.is_name_custom) channelName = channel?.name;
+    if (channel?.is_name_custom) {
+        channelName = channel?.name;
+    } else {
+        // assign default value or handle this case
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->